tools/CMakeLists.txt

```diff
@@ -54,27 +55,54 @@ endif()
 file(GLOB verovio_SRC "../src/*.cpp")
 file(GLOB midi_SRC "../src/midi/*.cpp")
 
-add_executable (
-    verovio
-    main.cpp
-    ${verovio_SRC}
-    ${hum_SRC}
-    ${midi_SRC}
-    ../src/pugi/pugixml.cpp
-    ../libmei/attconverter.cpp
-    ../libmei/atts_analytical.cpp
-    ../libmei/atts_cmn.cpp
-    ../libmei/atts_cmnornaments.cpp
-    ../libmei/atts_critapp.cpp
-    ../libmei/atts_gestural.cpp
-    ../libmei/atts_externalsymbols.cpp
-    ../libmei/atts_mei.cpp
-    ../libmei/atts_mensural.cpp
-    ../libmei/atts_midi.cpp
-    ../libmei/atts_pagebased.cpp
-    ../libmei/atts_shared.cpp
-    ../libmei/atts_visual.cpp
-)
+if (BUILD_AS_LIBRARY)
+    message(STATUS "build verovio as library")
+    add_library (
+         verovio
+         SHARED
+         ${verovio_SRC}
+         ${hum_SRC}
+         ${midi_SRC}
+             ../src/pugi/pugixml.cpp
```

Comment:
We have too much code duplication here. We need to find a way to have the list of common source files in a variable.

Reply:
Changed that!

In addition I've added a new custom target for the header files. That is required for QtCreator, otherwise all .h files are not shown in the project tree. The custom target should not have any consequences to other IDEs.